google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -1849,6 +1979,83 @@ def self.gapi_from_attrs gapi, attributes
           Google::Apis::StorageV1::Object.new(**attr_params)
         end
 
+        ##
+        # Represents a reference to a file, with optional preconditions.
+        # See `sources` in {Bucket#compose}.
```

Comment:
@quartzmo, recommendation: please split out Bucket#compose preconditions support into a separate follow-up FR. I suspect we will need to dig in a bit more on what the best way to represent this information will be. I don't want to jump to a specific design yet.

Reply:
Removed Bucket#compose preconditions support from this PR.

Reply:
Moved to #11690.